component/loki/process/internal/metric/counters.go

```diff
@@ -0,0 +1,115 @@
+package metric
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	CounterInc = "inc"
+	CounterAdd = "add"
+)
+
+// CounterConfig defines a counter metric whose value only goes up.
+type CounterConfig struct {
+	// Shared fields
+	Name         string        `river:"name,attr"`
+	Description  string        `river:"description,attr,optional"`
+	Source       string        `river:"source,attr,optional"`
+	Prefix       string        `river:"prefix,attr,optional"`
+	IdleDuration time.Duration `river:"max_idle_duration,attr,optional"`
+	Value        string        `river:"value,attr,optional"`
+
+	// Counter-specific fields
+	Action          string `river:"action,attr"`
+	MatchAll        bool   `river:"match_all,attr,optional"`
+	CountEntryBytes bool   `river:"count_entry_bytes,attr,optional"`
+}
+
+// DefaultCounterConfig sets the default for a Counter.
+var DefaultCounterConfig = CounterConfig{
+	IdleDuration: 5 * time.Minute,
+}
+
+// UnmarshalRiver implements the unmarshaller
+func (c *CounterConfig) UnmarshalRiver(f func(v interface{}) error) error {
+	*c = DefaultCounterConfig
+	type counter CounterConfig
+	err := f((*counter)(c))
+	if err != nil {
+		return err
+	}
+
+	if c.IdleDuration < 1*time.Second {
```

Comment:
Should this be `<=`?

Reply:
To mimic promtail, we want to allow the `"1s"` as a valid value.

Reply:
Then we should change the error message `"idle duration must be greater than 1s"`.